src/pages/ProfilesExplorerView/components/SceneExploreDiffFlameGraph/components/SceneComparePanel/components/SceneTimeRangeWithAnnotations.ts

```diff
@@ -0,0 +1,204 @@
+import { dateTime, TimeRange } from '@grafana/data';
+import {
+  sceneGraph,
+  SceneObjectBase,
+  SceneObjectUrlSyncConfig,
+  SceneObjectUrlValues,
+  SceneTimeRangeLike,
+  SceneTimeRangeState,
+  VizPanel,
+} from '@grafana/scenes';
+import { omit } from 'lodash';
+
+import { evaluateTimeRange } from '../domain/evaluateTimeRange';
+import { parseUrlParam } from '../domain/parseUrlParam';
+import { RangeAnnotation } from '../domain/RangeAnnotation';
+
+export enum TimeRangeWithAnnotationsMode {
+  ANNOTATIONS = 'annotations',
+  DEFAULT = 'default',
+}
+
+interface SceneTimeRangeWithAnnotationsState extends SceneTimeRangeState {
+  annotationTimeRange: TimeRange;
+  mode: TimeRangeWithAnnotationsMode;
+  annotationColor: string;
+  annotationTitle: string;
+}
+
+const TIMERANGE_NIL = {
+  from: dateTime(0),
+  to: dateTime(0),
+  raw: { from: '', to: '' },
+};
+
+export class SceneTimeRangeWithAnnotations
```

Comment:
Based on https://github.com/grafana/scenes/pull/744, thank you @darrenjaneczek !